modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -384,14 +384,33 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
         CV_WRAP Net();  //!< Default constructor.
         CV_WRAP ~Net(); //!< Destructor frees the net only if there aren't references to the net anymore.
 
-        /** @brief Create a network from Intel's Model Optimizer intermediate representation.
+        /** @brief Create a network from Intel's Model Optimizer intermediate representation (IR).
          *  @param[in] xml XML configuration file with network's topology.
          *  @param[in] bin Binary file with trained weights.
          *  Networks imported from Intel's Model Optimizer are launched in Intel's Inference Engine
          *  backend.
          */
         CV_WRAP static Net readFromModelOptimizer(const String& xml, const String& bin);
 
+        /** @brief Create a network from Intel's Model Optimizer in-memory buffers with intermediate representation (IR).
+         *  @param[in] bufferModelConfig buffer with model's configuration.
+         *  @param[in] bufferWeights buffer with model's trained weights.
+         *  @returns Net object.
+         */
+        CV_WRAP static
+        Net readFromModelOptimizerBuffer(const std::vector<uchar>& bufferModelConfig, const std::vector<uchar>& bufferWeights);
```

Comment:
All the versions should have the same name: readFromModelOptimizer and readNetFromModelOptimizer.

Reply:
The same name is not good and should be avoided.
At least for bindings support.
"String" (bytes array in meaning of UTF-8 file path) and `vector<uchar>` (or other forms of bytes array) is similar things and bindings wrappers may convert between them.

`readFromModelOptimizer` should be called `readFromModelOptimizerFile` but it is too late.

---

What is about prefixes?
- "read" for files ("load" is also used, for example in objdetect module)
- "create" / "build" for buffers

Reply:
Here an example when bindings work correctly with Darknet model: https://github.com/opencv/opencv/pull/11104#issuecomment-403763020

Reply:
I didn't say that they not working. My concern here is that overused overloading is not reliable.

Feel free to update patch (just keep eyes on documentation).

Reply:
@Quantizs, please change `Net::readFromModelOptimizer*` as well.